### PR TITLE
Fix the derivatives and improve anisotropic GrandPotential model

### LINKED ISSUE #12554: Mismatch between the anisotropic function and its derivatives in InterfaceOrientationMultiphaseMaterial
## Rationale

`InterfaceOrientationMultiphaseMaterial` uses two different functions for calculating the angle and its derivatives. This mismatch causes asymmetric precipitation growth in the GrandPotential Solidification model.

## Description
Visualizing the interfacial coefficients and its derivatives in the GrandPotential Solidification tests/examples would demonstrate the asymmetric behavior.

## Impact

Fixes the asymmetric behavior, improves the model.

### PR DESCRIPTION
**Pull request by SudiptaBiswas**
- Corrects the derivatives in the `InterfaceOrientationMultiphaseMaterial`
- Resolves the asymmetric growth issue, improves the anisotropic behavior 
- Improves the jacobians as well

Closes #12554

### CODE REVIEW

**Comment by SudiptaBiswas** [comment]
Proof of improved symmetry
symmetry_plot

**Comment by SudiptaBiswas** [comment]
Ping @laagesen @dschwen

File: modules/phase_field/examples/anisotropic_interfaces/GrandPotentialTwophaseAnisotropy.i [hunk 1]
```diff
@@ -7,13 +7,16 @@
...
+  uniform_refine = 2
 []
 
 [GlobalParams]
   radius = 0.5
   int_width = 0.3
   x1 = 0
   y1 = 0
+  derivative_order = 2
+  enable_jit = false
```
Comment (dschwen) [inline modules/phase_field/examples/anisotropic_interfaces/GrandPotentialTwophaseAnisotropy.i]: is this still necessary?

File: modules/phase_field/examples/anisotropic_interfaces/GrandPotentialTwophaseAnisotropy.i [hunk 2]
```diff
@@ -235,19 +234,20 @@
...
+    prop_values = '1.0 1.0  0.1  1.0  10.0  0.1  10.0  0.9   4.5 10.0'
   [../]
   [./Mobility]
     type = DerivativeParsedMaterial
     f_name = Dchi
     material_property_names = 'D chi'
     function = 'D*chi'
-    derivative_order = 2
-    enable_jit = false
+    derivative_order = 0
```
Comment (dschwen) [inline modules/phase_field/examples/anisotropic_interfaces/GrandPotentialTwophaseAnisotropy.i]: Just use `ParsedMaterial` here

File: modules/phase_field/src/materials/InterfaceOrientationMultiphaseMaterial.C [hunk 3]
```diff
@@ -85,52 +84,58 @@ InterfaceOrientationMultiphaseMaterial::computeQpProperties()
   if (n < -cutoff)
     n = -cutoff;
 
+  // Calculate the orientation angle
   const Real angle = std::acos(n) * MathUtils::sign(ny);
```
Comment (dschwen) [inline modules/phase_field/src/materials/InterfaceOrientationMultiphaseMaterial.C]: this should be rewritten using `atan2` (but you don't have to do that in this PR)

File: modules/phase_field/examples/anisotropic_interfaces/GrandPotentialTwophaseAnisotropy.i (hunk 1 above)
Comment (SudiptaBiswas) [reply modules/phase_field/examples/anisotropic_interfaces/GrandPotentialTwophaseAnisotropy.i]: Just to speed up a little.

Comment (dschwen) [reply modules/phase_field/examples/anisotropic_interfaces/GrandPotentialTwophaseAnisotropy.i]: No: (running with 4 MPI processes)

```
JIT=false
real	1m49.247s
user	7m5.080s
sys	0m10.293s

JIT=true (1st run)
real	1m21.971s
user	5m17.654s
sys	0m9.571s

JIT=true (2nd run)
real	1m20.879s
user	5m14.455s
sys	0m8.708s
```

Runs **over 30% faster with JIT**, and the JIT overhead (difference between 1st and 2nd run, where the cached functions are used) is only one second. It does not make sense to disable JIT here!

**Review by dschwen — CHANGES_REQUESTED** [review]

File: modules/phase_field/examples/anisotropic_interfaces/GrandPotentialTwophaseAnisotropy.i (hunk 1 above)
Comment (SudiptaBiswas) [reply modules/phase_field/examples/anisotropic_interfaces/GrandPotentialTwophaseAnisotropy.i]: I see. Okay, I will enable it then.

**Review by dschwen — APPROVED** [review]

**Comment by laagesen** [comment]
Dendrite morphology looks good, thanks @SudiptaBiswas.

**Pull request merged**